### PR TITLE
Add type annotations to Frame (generic.py)

### PR DESCRIPTION
**Pull request by xinrong-databricks**

### CODE REVIEW

File: databricks/koalas/generic.py [hunk 1]
```diff
@@ -375,7 +402,7 @@ def get_dtype_counts(self):
         if not isinstance(self.dtypes, Iterable):
             dtypes = [self.dtypes]
         else:
-            dtypes = self.dtypes
+            dtypes = cast(list, self.dtypes)
```
Comment (ueshin) [inline databricks/koalas/generic.py:405]: `pd.Series`?
Or maybe we can just make list here: `list(self.dtypes)`?

File: databricks/koalas/generic.py [hunk 2]
```diff
@@ -586,7 +613,7 @@ def to_csv(
         partition_cols: Optional[Union[str, List[str]]] = None,
         index_col: Optional[Union[str, List[str]]] = None,
         **options
-    ):
+    ) -> None:
```
Comment (ueshin) [inline databricks/koalas/generic.py:616]: `Optional[str]`? `to_csv` will return string when `path is None`.

File: databricks/koalas/generic.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/generic.py]: Makes sense! Modified.

File: databricks/koalas/generic.py (hunk 2 above)
Comment (xinrong-databricks) [reply databricks/koalas/generic.py]: Good catch! Updated.

File: databricks/koalas/generic.py [hunk 3]
```diff
@@ -640,6 +667,10 @@ def to_csv(
             It has higher priority and overwrites all other options.
             This parameter only works when `path` is specified.
 
+        Returns
+        -------
+        str or None
```
[the remark this answers is not in the document]
Comment (xinrong-databricks) [reply databricks/koalas/generic.py:672]: Sounds good!

File: databricks/koalas/generic.py [hunk 4]
```diff
@@ -375,7 +402,7 @@ def get_dtype_counts(self):
         if not isinstance(self.dtypes, Iterable):
             dtypes = [self.dtypes]
         else:
-            dtypes = self.dtypes
+            dtypes = list(self.dtypes)
         return pd.Series(dict(Counter([d.name for d in list(dtypes)])))
```
[the remark this answers is not in the document]
Comment (xinrong-databricks) [reply databricks/koalas/generic.py:406]: I'm afraid we will get
`databricks/koalas/generic.py:405: error: Incompatible types in assignment (expression has type "Iterable[Any]", variable has type "List[Any]")`
if L405 becomes `dtypes = self.dtypes`.

[the remark this answers is not in the document]
Comment (ueshin) [reply databricks/koalas/generic.py:406]: ah, I mean `list()` in `return pd.Series(dict(Counter([d.name for d in list(dtypes)])))`.
-> I guess `return pd.Series(dict(Counter([d.name for d in dtypes])))` should be fine.

[the remark this answers is not in the document]
Comment (xinrong-databricks) [reply databricks/koalas/generic.py]: Makes sense! Modified.

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**